recipes/LibriSpeech/librispeech_prepare.py

```diff
@@ -42,298 +50,317 @@ class LibriSpeechPreparer:
+    splits = splits
+    save_folder = save_folder
+    select_n_sentences = select_n_sentences
+    use_lexicon = use_lexicon
+    conf = {
+        "select_n_sentences": select_n_sentences,
+        "use_lexicon": use_lexicon,
+    }
+
+    # Other variables
+    # Saving folder
+    if not os.path.exists(save_folder):
+        os.makedirs(save_folder)
+
+    save_opt = os.path.join(save_folder, OPT_FILE)
+
+    if use_lexicon:
+        lexicon_dict = read_lexicon(
+            os.path.join(data_folder, "librispeech-lexicon.txt")
+        )
+
+    # Check if this phase is already done (if so, skip it)
+    if skip(splits, save_folder, conf):
+        logger.debug("Skipping preparation, completed in previous run.")
+        return
 
-    def __init__(
-        self,
-        data_folder,
-        splits,
-        save_folder,
-        select_n_sentences=None,
-        use_lexicon=False,
-    ):
-        super().__init__()
-        self.data_folder = data_folder
-        self.splits = splits
-        self.save_folder = save_folder
-        self.select_n_sentences = select_n_sentences
-        self.use_lexicon = use_lexicon
-        self.conf = {
-            "select_n_sentences": select_n_sentences,
-            "use_lexicon": use_lexicon,
-        }
-
-        # Other variables
-        self.samplerate = 16000
-        # Saving folder
-        if not os.path.exists(self.save_folder):
-            os.makedirs(self.save_folder)
-
-        self.save_opt = self.save_folder + "/opt_librispeech_prepare.pkl"
+    # Additional checks to make sure the data folder contains Librispeech
+    check_librispeech_folders()
 
-        if use_lexicon:
-            self.lexicon_dict = self.read_lexicon(
-                self.data_folder + "/librispeech-lexicon.txt"
-            )
+    # create csv files for each split
+    for split_index in range(len(splits)):
 
-        # Check if this phase is already done (if so, skip it)
-        if self.skip():
-            for split in self.splits:
-                text = self.save_folder + "/" + split + ".csv"
-                msg = "\t" + text + " created!"
-                logger.debug(msg)
-            return
+        split = splits[split_index]
 
-        # Additional checks to make sure the data folder contains Librispeech
-        self.check_librispeech_folders()
+        wav_lst = get_all_files(
+            os.path.join(data_folder, split), match_and=[".flac"]
+        )
 
-        # create csv files for each split
-        for split_index in range(len(self.splits)):
+        text_lst = get_all_files(
+            os.path.join(data_folder, split), match_and=["trans.txt"]
+        )
 
-            split = self.splits[split_index]
+        text_dict = text_to_dict(text_lst)
 
-            wav_lst = get_all_files(
-                self.data_folder + "/" + split, match_and=[".flac"]
-            )
+        if select_n_sentences is not None:
+            select_n_sentences = select_n_sentences[split_index]
```

Comment:
The documentation of this function says `select_n_sentences` is an `int`,
how can it support the index operator `[]`?

Reply:
You're right, the `select_n_sentences` variable is not used and should be removed. Feel free to open a PR for that!